pennylane/debugging/debugger.py

```diff
@@ -132,33 +132,198 @@ def pldb_device_manager(device):
 
 
 def breakpoint():
-    """Launch the custom PennyLane debugger."""
+    """A function which freezes execution and launches the PennyLane debugger (PLDB).
+
+    This function marks a location in a quantum circuit (QNode). When it is encountered during
+    execution of the quantum circuit, an interactive debugging prompt is launched to step
+    throught the circuit execution using Pdb like commands (:code:`list`, :code:`next`,
+    :code:`continue`, :code:`quit`).
+
+    **Example**
+
```

Comment:
Is it worth mentioning the `debug_x` functions we've added? E.g., with a note/see also box.

Reply:
I can add a see also link to the debugging module documentation, the example there covers the rest of the functions?  

OR, if you prefer I can list all of the debug_x functions and link to their individual documentation pages.